tests/sm_int/sm_int.c

```diff
@@ -1420,23 +1478,23 @@ static void sm_switches_from_state_to_created(SM_GO_TO_STATE* goToState)
 
 TEST_FUNCTION(STATE_and_API)

```

Comment:
>TEST_FUNCTION [](http://example.com/codeflow?start=0&length=13)

Could we add a test which does what we are trying to fix.

1. Start a thread that will sm_exec_begin and wait for some interlocked variable before calling sm_exec_end (wait forever until cancelled)
2. Call sm_close_begin_with_cb
3. The callback will signal the first thread to complete (this is "cancel")
...